hooks/firebird/use-event-stream.tsx

```diff
@@ -114,59 +122,39 @@ const useEventStream = (deploymentId: string, deploymentStatus?: string) => {
     [updateDeploymentProgress, updateDeploymentStatus],
   )
 
-  /**
-   * need to follow the appropriate event stream based on the instance action taking place (Start, Deploy, Stop)
-   * @returns method string for the event stream api call
-   */
-  const getMethod = (): string => {
-    switch (deploymentStatus) {
-      case DeploymentStatus.STARTING:
-        return "follow_deployment_start"
-      case DeploymentStatus.STOPPING:
-      case DeploymentStatus.STOPPED:
-        return "follow_deployment_stop"
-      case DeploymentStatus.DEPLOYING:
-        return "follow_deployment"
-      case DeploymentStatus.ACTIVE:
-        return lastDeploymentStatus === DeploymentStatus.STARTING
-          ? "follow_deployment_start"
-          : "follow_deployment"
-      default:
-        return "follow_deployment"
-    }
-  }
-
-  const method = getMethod()
+  const getMethod = useCallback(
+    (deploymentStatus: DeploymentStatus): string => {
+      switch (deploymentStatus) {
+        case DeploymentStatus.STARTING:
+          return "follow_deployment_start"
+        case DeploymentStatus.STOPPING:
+          return "follow_deployment_stop"
+        case DeploymentStatus.DEPLOYING:
+          return "follow_deployment"
+        case DeploymentStatus.PENDING:
+          return "follow_deployment"
+        default:
+          return "follow_deployment"
+      }
+    },
+    [],
+  )
```

Comment:
Why useCallback here?

Reply:
https://stackoverflow.com/questions/71265042/what-is-usecallback-in-react-and-when-to-use-it

TBH, useCallback is probably a bit contrived and an over-optimization here. But the other functions that martin defined and called in each hook used useCallback, so i just kept it consistent.

There may be a good argument not to use it in our functions at times: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
You probably know better than me what is appropriate here. I was just going for consistency.

Reply:
consistency is good